### PR TITLE
Xavier/ece220

### PR DESCRIPTION
**Pull request by xavierrouth**
hi guys :) 🐈

### CODE REVIEW

**Comment by xavierrouth** [comment]
I need someone who has taken this class more recently to help fill in the tips section / review in general. Most of the other content should be good.

**Comment by xavierrouth** [comment]
idk how to resolve conflict w 408 this. 408 on main is correct, this branch doens't amtter.

**Comment by xavierrouth** [comment]
idk how to fix this

File: docs/Course Wiki/ECE Course Offerings/ECE220.md [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+- Basic C++ programming
+
+ECE220 is meant to be a first-time introduction to higher level programming languages like C and C++. It is meant to prepare students for classes like [CS225](../CS%20Course%20Offerings/CS225.md) and [ECE391](ECE391.md) where a solid programming foundation is necessary. The first part of the course expands upon low-level LC-3 programming taught in ECE120. This covers TRAPs, interrupts, I/O, and calling conventions, all of which is also covered in ECE391.
+
+ECE220 then moves on to teaching C programming, often referencing LC-3. Students learn about pointers, recursion, arrays, memory management, and other fundamental computer programming concepts. The course also covers some simple data structures, such as binary search trees, linked lists, and dynamic arrrays. Most of the data structure content is meant to prepare students for CS225, where it will be covered in much more detail. Several basic sorting algorithms are also covered in the course.
+
+At the very end of the course, the class breifly touches on C++ and object-oriented design patterns, such as inheritance.
+
+## Prerequisites
+- [ECE120](ECE120.md)
```
Comment (rzig) [inline docs/Course Wiki/ECE Course Offerings/ECE220.md:25]: add name of ece120

File: docs/Course Wiki/ECE Course Offerings/ECE220.md [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+ECE220 also has multiple quizzes, held at the CBTF in between midterm exams. To prepare for these quizzes, students are recommended to go over past MPs and lecture material. Quizzes will consist of free-response auto-graded coding questions in either LC-3, C, or C++. They will typically be extremely similar to the topics covered in lectures in the past week or algorithms written in MPs. 
+
+Note that the course structure of ECE220 often varies slightly from semester to semester. For the most up to date information, view the syllabus or the course website.
+
+## Instructors
+This class is taught by a variety of professors, most recently these have included Yuting Wu Chen, Ivan Thomas Abraham, and Volodymr Kindratenko.
+
+## Course Tips
+Don't go to lecture or lab.
```
Comment (rzig) [inline docs/Course Wiki/ECE Course Offerings/ECE220.md]: so true

Comment (rzig) [reply docs/Course Wiki/ECE Course Offerings/ECE220.md]: but maybe explain why lecture/lab aren't that useful

File: docs/Course Wiki/ECE Course Offerings/ECE220.md [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+## Life After
+
+Students who enjoyed the first parts of ECE220 (Interrupts, I/O, Assembly / C Programming) will enjoy [ECE391](ECE391.md), as it covers most of the same things, but in much greater detail, with more interesting programming projects, and with real examples (x86 and the Linux Kernel instead of LC3).
+
+Students who enjoyed the later parts of ECE220 (Data Structures, Recursion, C++ Programming) will enjoy [CS225](CS225.md), as it covers most of the same things, but in much greater detail. Because of the overlap in topics, 
+
+Taking ECE 220 opens up some higher level courses to highly motivated undergrads. Keep in mind you need to have completed your [junior eligibility requirements](https://ece.illinois.edu/academics/ugrad/advising-tips/junior225) first. [ECE408- Applied Parallel Programming](ECE408.md) is one such course that only requires ECE 220 as a prerequisite.
```
Comment (rzig) [inline docs/Course Wiki/ECE Course Offerings/ECE220.md]: no space between ECE and 220

**Review by rzig — CHANGES_REQUESTED** [review]
few nitpicks otherwise lgtm

File: docs/Course Wiki/ECE Course Offerings/ECE220.md [hunk 4]
```diff
@@ -33,32 +33,32 @@ Computer Engineering students should aim to take this course immediatley after t
...
+Students who enjoyed the later parts of ECE220 (Data Structures, Recursion, C++ Programming) will enjoy [CS225](../CS%20Course%20Offerings/CS225.md), as it covers most of the same things, but in much greater detail. Because of the overlap in topics, 
 
-Taking ECE220 opens up some higher level courses to highly motivated undergrads. Keep in mind you need to have completed your [junior eligibility requirements](https://ece.illinois.edu/academics/ugrad/advising-tips/junior225) first. [ECE408- Applied Parallel Programming](ECE408.md) is one such course that only requires ECE 220 as a prerequisite.
+Taking ECE220 opens up some higher level courses (that aren't explicitly required) to highly motivated undergrads. Keep in mind you need to have completed your [junior eligibility requirements](https://ece.illinois.edu/academics/ugrad/advising-tips/junior225) first. [ECE408 - Applied Parallel Programming](ECE408.md) is one such course that only requires ECE 220 as a prerequisite. Some other courses include [CS 431 - Embedded Systems](../CS%20Course%20Offerings/CS431.md). If you liked translating C to LC3, look into [CS 426 - Compiler Construction.](../CS%20Course%20Offerings/CS426.md)
```
Comment (rzig) [inline docs/Course Wiki/ECE Course Offerings/ECE220.md]: maybe move course list to first sentence - "motivated undergrads, such as ECE408, CS431, ..."

File: docs/Course Wiki/ECE Course Offerings/ECE220.md [hunk 5]
```diff
@@ -33,32 +33,32 @@ Computer Engineering students should aim to take this course immediatley after t
...
 - C to LC3: Use compiler explorer! or read the textbook!
+- Runtime Stack: Use compiler explorer! or read the textbook!
 - Pointers: Pointers can be confusing to beginner programmers, however they are fundamental to understanding how computers work, and will be used alot in other required classes (CS 225, ECE 391).
 - Memory Management: Dynamic memory management is confusing, you will deal with this more in CS 225.
 - Strings: Strings are a great way to benchmark your understanding of C programming, if you can understand strings and their underlying implementation well, chances are you doing well in the course.
-- C++: The C++ section of the course is taught very fast, and can be confusing. Good luck!
+- C++: The C++ section of the course is taught very fast and can be confusing. Good luck!
 
 ## Resources
 - [Compiler Explorer](https://www.xavierrouth.com/lc3compiler)
```
Comment (rzig) [inline docs/Course Wiki/ECE Course Offerings/ECE220.md:64]: compiler explorer :star_struck:

**Review by rzig — CHANGES_REQUESTED** [review]

**Review by rzig — APPROVED** [review]

**Pull request merged**